src/fabric.c

```diff
@@ -131,6 +158,27 @@ static int fi_register_provider(struct fi_provider *provider, void *dlhandle)
 
 	prov->dlhandle = dlhandle;
 	prov->provider = provider;
+
+	/*
+	 * If context->internal[0] is not set then generate a position to use.
+	 * If this is the first one then use DYNAMIC_START, if not then check
+	 * the tail and add 1.
+	 */
+	if (!FI_LOG_CTX(provider)) {
+		FI_LOG_CTX(provider) = malloc(sizeof(int));
+		if (prov_tail) {
+			*(int *)FI_LOG_CTX(provider) =
+			    *(int *)FI_LOG_CTX(prov_tail->provider) + 1;
+		} else {
+			*(int *)FI_LOG_CTX(provider) = DYNAMIC_START;
+		}
+	}
```

Comment:
Yeah, same comment. Needs to be cleaned up. This was a quick proof of concept thing so I could get a sense of whether I was on the right direction...